src/hooks/sync-sites/use-site-sync-management.ts

```diff
@@ -37,6 +39,33 @@ export const useSiteSyncManagement = ( {
 		}
 	}, [ isAuthenticated, loadConnectedSites ] );
 
+	// whenever array of syncSites changes, we need to update connectedSites to keep them updated with wordpress.com
+	useEffect( () => {
+		if ( isFetching || ! isAuthenticated ) {
+			return;
+		}
+
+		setConnectedSites( ( prevConnectedSites ) => {
+			const updatedConnectedSites = prevConnectedSites.map( ( connectedSite ) => {
+				const site = syncSites.find( ( site ) => site.id === connectedSite.id );
+
+				if ( ! site ) {
+					return connectedSite;
+				}
+
+				return {
+					...connectedSite,
+					syncSupport: site.syncSupport,
+					url: site.url,
```

Comment:
In this PR, I wanted to introduce an up-to-date mechanism, which was crucial for the issues I was working on at that moment. And I planned to sync other fields later, since I wasn't aware enough about how we are working with staging sites, so wanted to spend time to do it carefully separately. So I will do it now :)

Reply:
@sejas  So, as I expected - up-to-dating staging sites is not easy process, as syncSupport and url :)

I have figured out how we work with them and carefully made all necessary changes, [PR is here](https://github.com/Automattic/studio/pull/700).

Since you have context and knowledge of this part of code, I think that it makes sense to ask specifically you about reviewing [the PR](https://github.com/Automattic/studio/pull/700). Also, I need your fresh pair of eyes :) Since synchronization is not easy process, and I tried to do it with a few different approaches, rewriting code a few time  and the current one looks optimal. So will be waiting for your review and thoughts. 
Thanks :)